### PR TITLE
Fix bug in dimensionality check

### PR DESCRIPTION
**Pull request by tdent**

### CODE REVIEW

File: popde/density_estimate.py [hunk 1]
```diff
@@ -24,21 +24,20 @@ def __init__(self, data, bandwidth=1.0, dim_names=None):
                         values must be strings
         """
         if len(data.shape) != 2:
-            raise ValueError("data must have shape (n_samples, n_features).")
-
+            raise ValueError("Data must have shape (n_samples, n_features).")
```
Comment (JAMSADIQ) [inline popde/density_estimate.py:27]: correct change.

File: popde/density_estimate.py [hunk 2]
```diff
@@ -24,21 +24,20 @@ def __init__(self, data, bandwidth=1.0, dim_names=None):
...
 
     def check_dimensionality(self):
         """
         Check if the dimension of training data matches the number of parameter names.
         """
-        if len(self.data.shape[0]) != len(self.dim_names):
-            raise ValueError("Dimensionality of data array does not match the number of dimension names.")
+        if self.data.shape[1] != len(self.dim_names):
+            raise ValueError("Dimensionality of data array does not match the
+                             " number of dimension names.")
```
Comment (JAMSADIQ) [inline popde/density_estimate.py]: ok

File: popde/density_estimate.py [hunk 3]
```diff
@@ -148,4 +148,4 @@ def estimation(self, samples, x_eval, alphagrid, bwgrid):
         self.optbw, self.optalpha = optvalues[0], optvalues[1]
         kdeval = train_eval_kde(samples, x_eval, self.optbw, self.optalpha)
 
-    return kdeval, self.optbw, self.optalpha 
+        return kdeval, self.optbw, self.optalpha
```
Comment (JAMSADIQ) [inline popde/density_estimate.py:151]: correct fix

File: popde/density_estimate.py [hunk 4]
```diff
@@ -36,8 +36,8 @@ def check_dimensionality(self):
         Check if the dimension of training data matches the number of parameter names.
         """
         if self.data.shape[1] != len(self.dim_names):
-            raise ValueError("Dimensionality of data array does not match the
-                             " number of dimension names.")
+            raise ValueError("Dimensionality of data array does not match "
+                             "the number of dimension names.")
```
Comment (JAMSADIQ) [inline popde/density_estimate.py:40]: ok

**Review by JAMSADIQ — APPROVED** [review]
ok

**Review by JAMSADIQ — APPROVED** [review]

**Comment by tdent** [comment]
Oh, somehow I missed that you approved the other PR first without any comment and I merged that, so now there is a minor conflict to fix.

**Pull request merged**